sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystem.java

```diff
@@ -388,59 +448,45 @@ BlobServiceClient getBlobServiceClient() {
         return this.blobServiceClient;
     }
 
-    private BlobServiceClient buildBlobServiceClient(String endpoint, Map<String, ?> config) {
+    private BlobServiceClient buildBlobServiceClient(String endpoint, AzureFileSystemConfig config) {
         BlobServiceClientBuilder builder = new BlobServiceClientBuilder()
                 .endpoint(endpoint);
 
         // Set the credentials.
-        if (config.containsKey(AZURE_STORAGE_SHARED_KEY_CREDENTIAL)) {
-            builder.credential((StorageSharedKeyCredential) config.get(AZURE_STORAGE_SHARED_KEY_CREDENTIAL));
-        } else if (config.containsKey(AZURE_STORAGE_SAS_TOKEN_CREDENTIAL)) {
-            builder.credential((AzureSasCredential) config.get(AZURE_STORAGE_SAS_TOKEN_CREDENTIAL));
+        if (config.getSharedKeyCredential() != null) {
+            builder.credential(config.getSharedKeyCredential());
+        } else if (config.getSasCredential() != null) {
+            builder.credential(config.getSasCredential());
```

Comment:
Do we want to do something similar as we do in the client builders where an exception is thrown when there are multiple forms credentials?

Reply:
It is already documented behavior that we take a credential preference instead of throwing when both. The library is not GA yet so that is still changeable, but I think this is beyond the scope of this PR.